src/Endpoint/FileEndpoint.php

```diff
@@ -23,39 +22,28 @@ class FileEndpoint extends EndpointBase implements TokenAwareInterface
     use TokenAwareTrait;
     use ValidationAwareTrait;
 
-    protected string $permanentId;
-
-    public function getPermanentId(): string
+    /**
+     * @inheritDoc
+     */
+    public function __construct()
```

Comment:
I dont lile it, I think this shows that this asset download is not part of the Endpoint architecture.
Now of course all our logic regarding making requests is in the EndpointBase class, which now makes it more clear that that is wrong. I would resolve D8AGE-393 first, after move this class out of the Endpoint space.

Reply:
Yes, it turned out that the architecture is not fully compatible with the API. I tried to make it cleaner, but it is not possible without a major refactor.